tests/bottle/test_vulnerable.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from bottle import Bottle
+from vulnpy.bottle.vulnerable_routes import add_vulnerable_routes
+from vulnpy.trigger import DATA
+from tests import parametrize_root, parametrize_triggers
+from webtest import TestApp
+
+
+@pytest.fixture(scope="module")
+def client():
+    app = Bottle(__name__)
+    add_vulnerable_routes(app)
+
+    class StripPathMiddleware(object):
+        def __init__(self, app):
+            self.app = app
+
+        def __call__(self, e, h):
+            e["PATH_INFO"] = e["PATH_INFO"].rstrip("/")
+            return self.app(e, h)
+
+    return TestApp(StripPathMiddleware(app))
+
+
+@parametrize_root
+def test_root_views(client, view_path):
+    response = client.get(view_path)
+    assert response.status_code == 200
+
+
+@parametrize_triggers
+@pytest.mark.parametrize("request_method", ["get", "post"])
+def test_trigger(client, request_method, view_name, trigger_name):
+    get_or_post = getattr(client, request_method)
+
+    data = DATA[view_name]
+
+    if view_name == "unsafe_code_exec":
+        data = "'{}'".format(data)
+
+    print(data)
+    response = get_or_post(
+        "/vulnpy/{}/{}/?user_input={}".format(view_name, trigger_name, data),
```

Comment:
if you remove the trailing `/` from the path here (the one right before `?`) you should be able to also get rid of the `StripPathMiddleware` in the app you're using for each test case, which would be kind of nice

Reply:
Yep you were right. It worked thanks.